### PR TITLE
refactor(query): if projection not query stat/owner will not get table_stats/ownership when access system.tables 

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Optimize system.tables speed

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [x] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16183)

### CODE REVIEW

File: src/query/storages/system/src/tables_table.rs [hunk 1]
```diff
@@ -247,8 +249,21 @@ where TablesTable<T, U>: HistoryAware
         let mut tables_ids: Vec<u64> = Vec::new();
         let mut db_name: Vec<String> = Vec::new();
 
+        let mut get_stats = true;
+        let stats_fields_indexes = [12, 13, 14, 15, 16, 17];
```
Comment (BohuTANG) [inline src/query/storages/system/src/tables_table.rs:253]: This seems not safe?

Comment (TCeason) [reply src/query/storages/system/src/tables_table.rs:253]: > let stats_fields_indexes = [12, 13, 14, 15, 16, 17];

this index is hard code by table schema If table schema modify the logic test will failed.

Comment (b41sh) [reply src/query/storages/system/src/tables_table.rs:253]: better to use field names to check, column index may change in the future.

File: src/query/storages/system/src/tables_table.rs [hunk 2]
```diff
@@ -409,7 +446,7 @@ where TablesTable<T, U>: HistoryAware
                             catalogs.push(ctl_name.as_str());
                             databases.push(db_name.to_owned());
                             database_tables.push(table);
-                            if ownership.is_empty() {
+                            if ownership.is_empty() || !get_ownership {
```
Comment (b41sh) [inline src/query/storages/system/src/tables_table.rs:449]: `!get_ownership` can remove, if `get_ownership` is false, `ownership` is always empty.

File: src/query/storages/system/src/tables_table.rs [hunk 3]
```diff
@@ -247,8 +249,39 @@ where TablesTable<T, U>: HistoryAware
         let mut tables_ids: Vec<u64> = Vec::new();
         let mut db_name: Vec<String> = Vec::new();
 
+        let mut get_stats = true;
+        let mut get_ownership = true;
+        let mut owner_field_indexes: Vec<usize> = Vec::new();
```
Comment (b41sh) [inline src/query/storages/system/src/tables_table.rs:254]: better to use `HashSet`

Comment (TCeason) [reply src/query/storages/system/src/tables_table.rs]: Done

File: src/query/storages/system/src/tables_table.rs (hunk 2 above)
Comment (TCeason) [reply src/query/storages/system/src/tables_table.rs]: Done

**Review by b41sh — APPROVED** [review]

**Pull request merged**